File: apphub-frontend/src/modules/utils/sql_generator/model/Segment.js
import Utils from "../../../../common/js/Utils";

const Segment = class {
    constructor(segmentType, parent = null, order = 0, value = null, id = Utils.generateRandomId()) {
        this.segmentType = segmentType;
        this.order = order;
        this.value = value;
        this.parent = parent;
        this.id = id;
    }
}

export default Segment;

File: apphub-service/apphub-service-utils/src/main/java/com/github/saphyra/apphub/service/utils/sql_generator/dao/segment/Segment.java
package com.github.saphyra.apphub.service.utils.sql_generator.dao.segment;

import com.github.saphyra.apphub.api.utils.model.sql_generator.SegmentType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;

import java.util.UUID;

@AllArgsConstructor
@Data
@Builder
public class Segment {
    private final UUID segmentId;
    private final UUID userId;
    private final UUID externalReference;
    private final SegmentType segmentType;
    private final Integer order;
}
